cmd/context/kubetoken.go

```diff
@@ -0,0 +1,146 @@
+	kc.staticKubetokenWarner.warn()
+
+	if okCtx == nil || okCtx.UserID == "" || okCtx.Cfg == nil || okCtx.Cfg.AuthInfos == nil || okCtx.Cfg.AuthInfos[okCtx.UserID] == nil {
+		return nil
+	}
+	okCtx.Cfg.AuthInfos[okCtx.UserID].Exec = nil
+	return nil
+}
+
+func (kc *staticKubetokenController) updateOktetoContextToken(uCtx *types.UserContext) error {
+	kc.staticKubetokenWarner.warn()
+	if uCtx.Credentials.Token == "" {
+		return errors.New("static kubernetes token not available")
+	}
+	return nil
+}
+
+// dynamicKubetokenController is used to update the kubeconfig stored in the okteto context by using a dynamic token to connect to the cluster.
+type dynamicKubetokenController struct {
+	oktetoClientProvider oktetoClientProvider
+}
+
+// newDynamicKubetokenController creates a new dynamicKubetokenController
+func newDynamicKubetokenController(okClientProvider oktetoClientProvider) *dynamicKubetokenController {
+	return &dynamicKubetokenController{
+		oktetoClientProvider: okClientProvider,
+	}
+}
+
+// updateOktetoContext when the dynamic kubetoken is enabled configures the kubeconfig auth method to be executed by okteto kubetoken
+// which returns a dynamic token to connect to the cluster.
+func (dkc *dynamicKubetokenController) updateOktetoContextExec(okCtx *okteto.OktetoContext) error {
+	okClient, err := dkc.oktetoClientProvider.Provide()
+	if err != nil {
+		return err
+	}
+
+	err = okClient.Kubetoken().CheckService(okCtx.Name, okCtx.Namespace)
+	if err != nil {
+		return fmt.Errorf("error checking kubetoken service: %w", err)
+	}
+
+	k8sCfg := okCtx.Cfg
+	if k8sCfg.AuthInfos == nil {
+		k8sCfg.AuthInfos = clientcmdapi.NewConfig().AuthInfos
+		k8sCfg.AuthInfos[okCtx.UserID] = clientcmdapi.NewAuthInfo()
+	}
+
+	if token := k8sCfg.AuthInfos[okCtx.UserID].Token; token != "" {
+		k8sCfg.AuthInfos[okCtx.UserID].Token = ""
+	}
+
+	k8sCfg.AuthInfos[okCtx.UserID].Exec = &clientcmdapi.ExecConfig{
+		APIVersion:         "client.authentication.k8s.io/v1",
+		Command:            "okteto",
+		Args:               []string{"kubetoken", "--context", okCtx.Name, "--namespace", okCtx.Namespace},
+		InstallHint:        "Okteto needs to be installed in your PATH and it has to be connected to your instance using the command 'okteto context use https://okteto.example.com'. Please visit https://www.okteto.com/docs/getting-started/ for more information.",
+		InteractiveMode:    "Never",
+		ProvideClusterInfo: true,
+	}
+	return nil
+}
+
+// updateOktetoContextToken retrieves a dynamic token for the given userContext and updates Credentials.Token
+// if error while retrieving the dynamic token or flag OKTETO_USE_STATIC_KUBETOKEN is enabled, value is not updated
+// and fallback to static token
+func (dkc *dynamicKubetokenController) updateOktetoContextToken(userContext *types.UserContext) error {
+	if userContext.User.Namespace == "" {
+		return errors.New("user context namespace is empty")
```

Comment:
should we add a fall back message about using static tokens? following the previous approach in this point

Reply:
It's already added on the function that is calling this one to keep the current behavior. I opted to put it on the upper function given that is the one that will doo the fallbback